### DISCUSSION TITLE
<script setup> with NodeViewWrapper

### DISCUSSION THREAD

**Discussion by jloganolson** [ueberdosis/tiptap#1825] [General Questions] [answered]
Anybody get Vue's new <script setup> working with NodeViewWrapper? I get the error "Cannot read property 'expose' of undefined" (and the editor fails to show content) when I try a very simple SFC:
```
<template>
  <NodeViewWrapper>YOOO</NodeViewWrapper>
</template>

<script setup>
import { NodeViewWrapper } from "@tiptap/vue-3";
</script>

<style lang="scss">
</style>
```

**Comment by petertoth** [comment, marked as answer]
@jloganolson I've got `<script setup>` syntax working with Tiptap Node Views in my vite project after I ran `yarn upgrade-interactive` So this issue seems to be solved now

**Reply by jaystey** [reply]
@petertoth Would you be able to give a simple example that is working for you? I'm getting the error "Cannot read properties of undefined (reading 'value')" even after running `yarn upgrade-interactive`, both on my own Vite project which uses node views and while testing jloganolson's SFC from above.

**Reply by petertoth** [reply]
@jaystey here it is https://stackblitz.com/edit/vitejs-vite-9gjgay?file=src/components/HelloWorld.vue

There's one more caveat:
When you use `<script setup lang="ts">` syntax with Typescript you will most likely want to have your node view props typed so you'd do something like:
```ts
import { NodeViewProps } from '@tiptap/vue-3'

const props = defineProps<NodeViewProps>()
```
However this results in this error: `[@vue/compiler-sfc] type argument passed to defineProps() must be a literal type, or a reference to an interface or literal type.`

There's a fantastic [Vite plugin](https://github.com/wheatjs/vite-plugin-vue-type-imports) by @wheatjs that'd fix that for you however I found it there's one more trick you need to do and that is to re-export it from your own file like:
```ts
// types.ts
import { NodeViewProps as TiptapNodeViewProps } from '@tiptap/core';

export interface NodeViewProps extends TiptapNodeViewProps {}
```
and when you import it then in your Vue Node Views it should work as expected.

cc @wheatjs perhaps you have an idea why is that ^ happening?

**Comment by volarname** [comment]
it will be great to document script setup examples, with some attributes also
